### DISCUSSION TITLE
Who wants this for websites?

### DISCUSSION THREAD

**Discussion by AVeryTemporaryUser** [discordjs/discord.js#8656] [Ideas]
Upvote this if you want djs for websites as well!

**Comment by Jiralite** [comment]
I'm pretty sure Discord no longer allows API requests from the browser from a bot, so this would not even be possible.

**Reply by AVeryTemporaryUser** [reply]
why does it not?

**Reply by NickantX** [reply]
Security, this would require exposing your bot token to all visitors of your website which poses a huge security risk.

**Reply by AVeryTemporaryUser** [reply]
ah, thanks for that info. i think its possible to access any sql database you make using javascript, if i put the token in a database and accessed it from there would it still pose a security threat?

**Reply by NickantX** [reply]
Your frontend should not be given direct, privileged access to a backend of any kind, be it a bot token, API key, or database credentials of any kind. Create a REST API that your frontend talks to. How you choose to create this API is up to you, if you wanted something cheap (free for small projects) and easy you could look into using Cloudflare Workers for it.

**Reply by AVeryTemporaryUser** [reply]
what if i made the discord bot in node.js and had it access the sql database for certain info that i want to share between the bot and the website, would that work? or is there something better that i can do to have them share info?

**Reply by NickantX** [reply]
As long as your frontend is not directly accessing your Discord bot token or database credentials, you'll be fine. How you choose to design your backend is up to you and is mostly a matter of preference.

**Reply by AVeryTemporaryUser** [reply]
thanks for the info
